pyquil/api/_qpu.py

```diff
@@ -71,34 +73,66 @@ def _extract_bitstrings(ro_sources: List[Optional[Tuple[int, int]]],
 
 class QPU(QAM):
     @_record_call
-    def __init__(self, endpoint: str, user: str = "pyquil-user", priority: int = 1) -> None:
+    def __init__(self,
+                 endpoint: Optional[str] = None,
+                 user: str = "pyquil-user",
+                 priority: int = 1,
+                 config: Optional[PyquilConfig] = None) -> None:
```

Comment:
might as well make this keyword-only like we're doing for `session` in the compiler file